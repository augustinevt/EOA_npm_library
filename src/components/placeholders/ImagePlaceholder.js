import React from 'react';
import Icon from '../icons';

class ImagePlaceholder extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="image-placeholder">
        <Icon type="image" />
        <div className="image-placeholder__label">
          Image Not Available
        </div>
      </div>
    );
  }
}

export default ImagePlaceholder;
