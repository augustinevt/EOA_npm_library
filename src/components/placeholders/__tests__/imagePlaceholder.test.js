import React from 'react';
import renderer from 'react-test-renderer';

import ImagePlaceholder from '../ImagePlaceholder';

test('When all props are valid', () => {
  const snapshot = renderer.create(
    <ImagePlaceholder />
  ).toJSON();

  expect(snapshot).toMatchSnapshot();
});
