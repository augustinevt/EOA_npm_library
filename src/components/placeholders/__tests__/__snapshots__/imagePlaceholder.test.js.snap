// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When all props are valid 1`] = `
<div
  className="image-placeholder"
>
  <div
    className="icon-wrapper"
  >
    <svg
      style={
        Object {
          "height": undefined,
          "overflow": "hidden",
        }
      }
      version="1.1"
      viewBox="0 0 1024 896"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M1024,576V896H0V704L192,512,320,640,704,256ZM192,384q-80,0-136-56T0,192Q0,112,56,56T192,0q80,0,136,56t56,136q0,80-56,136T192,384Z"
        />
      </g>
    </svg>
  </div>
  <div
    className="image-placeholder__label"
  >
    Image Not Available
  </div>
</div>
`;
